apps/nowcasting-app/components/map/pvLatest.tsx

```diff
@@ -32,42 +32,39 @@ const PvLatestMap = () => {
       refreshInterval: 1000 * 60 * 5, // 5min,
     },
   );
-  const selectedForcastValue = useMemo(() => {
-    if (!initForecastData || !selectedISOTime) return latestForecastValue;
 
-    const index = initForecastData.forecasts[0].forecastValues.findIndex(
-      (fv) => formatISODateString(fv.targetTime) === formatISODateString(selectedISOTime),
-    );
-    if (index >= 0) return index;
-    return latestForecastValue;
-  }, [initForecastData, selectedISOTime]);
-
-  const generateGeoJsonForecastData = (forecastData?: FcAllResData, forecastIndex?: number) => {
+  const generateGeoJsonForecastData = (forecastData?: FcAllResData, targetTime?: string) => {
     // Exclude first item as it's not represting gsp area
     const filteredForcastData = forecastData?.forecasts?.slice(1);
     const gspShapeDatat = gspShapeData as GeoJSON.FeatureCollection<GeoJSON.Geometry>;
     const forecastGeoJson = {
       ...gspShapeDatat,
-      features: gspShapeDatat.features.map((featureObj, index) => ({
-        ...featureObj,
-        properties: {
-          ...featureObj.properties,
-          expectedPowerGenerationMegawatts:
+      features: gspShapeDatat.features.map((featureObj, index) => {
+        let selectedFCvalue = filteredForcastData
```

Comment:
can you move this part as else statement for this condition if (targetTime)
` const selectedFCvalue =
          filteredForcastData && targetTime
            ? filteredForcastData[index]?.forecastValues.find(
                (fv) => formatISODateString(fv.targetTime) === formatISODateString(targetTime),
              )
            : filteredForcastData
            ? filteredForcastData[index]?.forecastValues[latestForecastValue]
            : undefined;`